### PR TITLE
Feature/unpeg

### PR DESCRIPTION
**Pull request by Lauraducky**

### CODE REVIEW

File: src/PockyBot.NET/Services/Triggers/Unpeg.cs [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+            return Task.FromResult(new Message
+            {
+                MessageParts = response
+            });
+        }
+
+        private MessagePart[] SendRandomResponse(Person sender, bool recipientIsPerson, string recipientName,
+            string recipientId, string roomId)
+        {
+            var random = _randomnessHandler.Random.Next(7);
```
Comment (jimcullenaus) [inline src/PockyBot.NET/Services/Triggers/Unpeg.cs:81]: I wonder if something like [the solution proposed in this SO answer](https://stackoverflow.com/a/34279195) might be neater?

File: src/PockyBot.NET/Services/Triggers/Unpeg.cs [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+            switch (random)
+            {
+                case 0:
+                    SendQueuedMessage(new Message {Text = "Kidding!", RoomId = roomId});
+                    return GetRecipientPegRemovedResponse(recipientIsPerson, recipientName, recipientId);
```
Comment (jimcullenaus) [inline src/PockyBot.NET/Services/Triggers/Unpeg.cs:86]: So if I'm understanding this correctly, we first queue up the second message, and *then* return the immediate response message?

I like the fact that this means we're not waiting around inside the responder thread, but it feels sort of awkward. Is there a way to return nothing at all, and instead queue up both options such that the first happens immediately and the second happens on a delay?

File: src/PockyBot.NET/Services/Triggers/Unpeg.cs (hunk 1 above)
Comment (Lauraducky) [reply src/PockyBot.NET/Services/Triggers/Unpeg.cs]: Good idea. done

File: src/PockyBot.NET/Services/Triggers/Unpeg.cs (hunk 2 above)
Comment (Lauraducky) [reply src/PockyBot.NET/Services/Triggers/Unpeg.cs:86]: Sure there is, but I'm not sure there's enough value in it to bother changing it.

File: src/PockyBot.NET/Services/UsernameUpdater.cs [hunk 3]
```diff
@@ -35,7 +35,7 @@ public async Task<List<PockyUser>> UpdateUsernames(List<PockyUser> users)
                 return u;
             }).ToList();
 
-            await Task.WhenAll(dbUpdates);
+            await Task.WhenAll(dbUpdates).ConfigureAwait(false);
```
[the remark this answers is not in the document]
Comment (Lauraducky) [reply src/PockyBot.NET/Services/UsernameUpdater.cs:38]: https://devblogs.microsoft.com/dotnet/configureawait-faq/

File: src/PockyBot.NET/Services/Triggers/Unpeg.cs (hunk 2 above)
Comment (jimcullenaus) [reply src/PockyBot.NET/Services/Triggers/Unpeg.cs:86]: I dunno, I sorta think of this one as the main point. Something feels *really* wrong to me about the idea of returning a response which is meant to be temporary and then later return a different response based on a timer from a helper task.

**Pull request merged**